pkgs/servers/web-apps/netbox/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib, pkgs, applyPatches, fetchFromGitHub, python3, plugins }:
+
+let
+  py = python3.override {
+    packageOverrides = self: super: {
+      django = super.django_3;
+      graphql-core = super.graphql-core.overridePythonAttrs (old: rec {
+        version = "3.1.7";
+        src = fetchFromGitHub {
+          owner = "graphql-python";
+          repo = old.pname;
+          rev = "v${version}";
+          sha256 = "1mwwh55qd5bcpvgy6pyliwn8jkmj4yk4d2pqb6mdkgqhdic2081l";
+        };
+      });
+      jsonschema = super.jsonschema.overridePythonAttrs (old: rec {
+        version = "3.2.0";
+        src = self.fetchPypi {
+          pname = old.pname;
+          inherit version;
+          sha256 = "c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a";
+        };
+      });
+    };
+  };
+
+  pyEnv = py.buildEnv.override {
+    extraLibs = with py.pkgs; [
+      django_3
+      django-cors-headers
+      django-debug-toolbar
+      django-filter
+      django-graphiql-debug-toolbar
+      django-mptt
+      django-pglocks
+      django-prometheus
+      django-redis
+      django-rq
+      django-tables2
+      django-taggit
+      django-timezone-field
+      djangorestframework
+      drf-yasg
+      graphene-django
+      gunicorn
+      jinja2
+      markdown
+      markdown-include
+      mkdocs-material
+      netaddr
+      pillow
+      psycopg2
+      pyyaml
+      social-auth-core
+      social-auth-app-django
+      svgwrite
+      tablib
+      jsonschema
+    ] ++ plugins;
+  };
+in
+  pkgs.symlinkJoin rec {
```

Comment:
Instead of a `symlinkJoin` please use `buildPythonApplication` with `format = "other"` and use `propagatedBuildInputs` like you normally would. Then expose the python path through passthru by calling `python3.pkgs.makePythonPath` on the `propgagatedBuildInputs`. That way you can treat it looks and feels more like a normal python application.

I think [paperless-ng](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/paperless-ng/default.nix) is a good example on how to package a Django application.